community/modules/scheduler/schedmd-slurm-gcp-v5-controller/main.tf

```diff
@@ -36,12 +36,14 @@ locals {
   enable_public_ip_access_config = var.disable_controller_public_ips ? [] : [{ nat_ip = null, network_tier = null }]
   access_config                  = length(var.access_config) == 0 ? local.enable_public_ip_access_config : var.access_config
 
+
   # Handle VM image format from 2 sources, prioritize source_image* variables
   # over instance_image
-  source_image_input_used = var.source_image != "" || var.source_image_family != "" || var.source_image_project != ""
-  source_image            = local.source_image_input_used ? var.source_image : lookup(var.instance_image, "name", "")
-  source_image_family     = local.source_image_input_used ? var.source_image_family : lookup(var.instance_image, "family", "")
-  source_image_project    = local.source_image_input_used ? var.source_image_project : lookup(var.instance_image, "project", "")
+  source_image_input_used         = var.source_image != "" || var.source_image_family != "" || var.source_image_project != ""
+  source_image                    = local.source_image_input_used ? var.source_image : lookup(var.instance_image, "name", "")
+  source_image_family             = local.source_image_input_used ? var.source_image_family : lookup(var.instance_image, "family", "")
+  source_image_project            = local.source_image_input_used ? var.source_image_project : lookup(var.instance_image, "project", "")
+  source_image_project_normalized = local.source_image != "" || strcontains(local.source_image_project, "/") ? local.source_image_project : "projects/${local.source_image_project}/global/images/family"
```

Comment:
nit: This block is identical in 3 different modules. Sometimes we try to capture this logic in a separate file so we can prevent drift. 

Example: [threads_per_core_calc.tf](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/modules/compute/vm-instance/threads_per_core_calc.tf) is found in two different modules but is forced to be maintained as identical via [duplicate-diff.py](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/d6a3ef45165c0c925b77b2e8eeddc9f69c35c22b/tools/duplicate-diff.py#L45-L46).

Reply:
Done.